File: backend/src/main/java/at/ac/htl/features/gpu/GPUMapper.java
package at.ac.htl.features.gpu;

import jakarta.enterprise.context.ApplicationScoped;

@ApplicationScoped
public class GPUMapper {
    public GPUDto toResource(GPU gpu) {
        return new GPUDto(
                gpu.gpu_id,
                gpu.name,
                gpu.price,
                gpu.chipset,
                gpu.memory,
                gpu.core_clock,
                gpu.boost_clock,
                gpu.color,
                gpu.length,
                gpu.img,
                gpu.model
        );
    }
}

File: android_frontend/app/src/main/java/at/htl/leonding/android_frontend/ui/screens/components/mapper/GpuMapper.kt
package at.htl.leonding.android_frontend.ui.screens.components.mapper

import at.htl.leonding.android_frontend.data.model.GpuDto
import at.htl.leonding.android_frontend.ui.screens.components.ComponentListItem

fun GpuDto.toListItem() = ComponentListItem(
    id = this.id,
    title = this.name,
    subtitle = "${this.chipset} • ${this.memory} GB GDDR6 • ${this.boostClock} MHz Boost",
    imageUrl = this.img,
    type = "gpu"
)